### DISCUSSION TITLE
onnx custom operator

### DISCUSSION THREAD

**Discussion by harshamedu** [onnx/onnx#4062] [Q&A]
Hi,

I am looking for a similar feature as keras custom layer in ONNX/Onnxruntime. The way I understand is to implement a custom operator in onnx for experimentation.  
The documentation seems to be pointing to implementation in C++ as a shared library and use it in python.
https://onnxruntime.ai/docs/reference/operators/add-custom-op.html

Is there a method to define custom op in python for onnx just for experimental purpose and use it for inferencing ? 
I tried following this but gives 'error: PyOp is not a registered function/op'
https://onnxruntime.ai/docs/reference/operators/custom-python-operator.html

**Comment by TruscaPetre** [comment]
Hello @harshamedu, I was playing with custom ops in the past month and I suggest you looking also at onnxruntime-extensions, there are a few other examples for implementing custom ops here https://github.com/microsoft/onnxruntime-extensions/tree/main/tutorials
